contents/javascript/VueJs/Intro.md

```diff
@@ -0,0 +1,172 @@
+# VueJs
+
+Author: [Lu Lechuan](https://github.com/LuLechuan)
+
+VueJs (also known as Vue) is an open-source progressive [JavaScript framework](https://en.wikipedia.org/wiki/JavaScript_framework) for building user interfaces. It is designed to improve code quality and maintainablity. Simply put, it is used as a tool to write JavaScript fast and easy.
+
+## Advantages of VueJs
+
+VueJs is an approachable, versatile and performant framework that helps to create a more maintainable and testable interface.
+
+#### Approachable
+
+VueJs is very easy to learn. Compared to other framework such as Augular, VueJs is simple in term of API and design. Learning enough to build non-trivial applications typically takes less than a day.
+
+#### Progressive
+
+VueJs is designed from the ground up to be incrementally adoptable. The core library is focused on the view layer only, and is easy to pick up and integrate with other libraries or existing projects. This means that if you have a large application, you can plug VueJs into just a part of your application without disturbing the other components. A quote from Evan You - the owner of VueJs is as follow:
+
+> Vue.js is a more flexible, less opinionated solution ( than Angular ). That allows you to structure your app the way you want it to be, instead of being forced to do everything the Angular way ( Angular requires a certain way to structure an application, making it hard to introduce Angular into an already built project ). It’s only an interface layer so you can use it as a light feature in pages instead of a full blown SPA ( single-page application ).
+
+#### Versatile
+
+VueJs is perfectly capable of powering sophisticated single-page applications when used in combination with modern tooling and supporting libraries.
+
+#### Clean
+VueJs syntax is simple and this can make the HTML pages very clean.
+
+#### Other Advantages of VueJs
+[What is VueJs and What are its Advantages](https://hackernoon.com/what-is-vue-js-and-what-are-its-advantages-4071b7c7993d)
+
+Detailed comparison of VueJs with other JavaScript frameworks can be found from [Vue Guild: Comparison with Other Frameworks](https://vuejs.org/v2/guide/comparison.html).
+
+## Drawbacks of VueJS
+
+#### Relatively small size community
+
+VueJs is a relatively new JavaScript framework as compared to Augular and React. The size of the community for VueJs is therefore relatively small. Although small size community means you can differentiate yourself from other JavaScript developers, it also means there are fewer resources such as tutorials and problem-shooting guides.
+
+#### Language barriers
+
+A majority of users of VueJs are the Chinese as VueJs is developed by a Chinese American. He is very supportive for the Chinese community and hence a lot of the existing plugins are written in Chinese. There might be some language barriers if an English speaking developer seeking for VueJs resources.
```

Comment:
Actually, I want to say he is supportive of the Chinese community coz that is the reason why he developed those plugins in Chinese so that Chinese developer can use. Is that right?